src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -114,6 +115,43 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     return this.resolveValue(value, type);
   }
 
+  getWithBrowserSettings<T = any>(
+    key: string,
+    defaultOverride?: T
+  ): {
+    advancedSettingValue: T | undefined;
+    browserValue: T | undefined;
+    defaultValue: T;
+  } {
+    const declared = this.isDeclared(key);
+
+    if (!declared && defaultOverride === undefined) {
+      throw new Error(
+        `Unexpected \`IUiSettingsClient.getWithBrowserSettings("${key}")\` call on unrecognized configuration setting "${key}".`
+      );
+    }
+
+    const type = this.cache[key]?.type;
+    const advancedSettingValue = this.cache[key]?.userValue;
+    const browserValue = this.browserStoredSettings[key]?.userValue;
+    const defaultValue = defaultOverride !== undefined ? defaultOverride : this.cache[key]?.value;
+
+    // Resolve the value based on the type
+    const resolveValue = (val: any): T => this.resolveValue(val, type);
+
+    const resolvedAdvancedValue =
+      advancedSettingValue !== undefined ? resolveValue(advancedSettingValue) : undefined;
```

Comment:
nit: could create a helper function to handle this undefined case and inline these with return